### PR TITLE
bugfix: empty tensor deallocation

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/disc/tests/regression/empty_tensor.cc [hunk 1]
```diff
@@ -45,4 +45,18 @@ TEST(EmptyTensor, ImplicitBroadcast2DF32) {
...
+// Corner case
+// Empty reshape + matmul
+TEST(EmptyTensor, EmptyReshapeMatmul) {
+  EXPECT_TRUE(feature_test_main(
+      /*mlir_file_path*/ c_ft_path + "empty_tensor_reshape_matmul.mlir",
+      /*backend_types*/
+      kSupportedBackendList,
+      /*num_inputs*/ 3,
+      /*num_outputs*/ 1,
+      /*input_descriptors*/ {"0x2x3xf32_X", "6x11xf32_X", "2xi32_h"},
```
Comment (minminsun) [inline tao_compiler/mlir/disc/tests/regression/empty_tensor.cc:57]: In which case a dim size could be 0?

File: tao_compiler/mlir/xla/ral/context/base/cpu/cpu_context_impl.cc [hunk 2]
```diff
@@ -151,6 +151,11 @@ buffer_t ral_base_cpu_alloc_persistent(ExecutionContext* ctx, size_t bytes) {
 }
 
 void ral_base_cpu_dealloc(ExecutionContext* ctx, buffer_t buffer) {
+  if (!buffer) {
+    TAO_VLOG(1) << "ral_base_cpu_dealloc early return for nullptr";
```
Comment (Yancey1989) [inline tao_compiler/mlir/xla/ral/context/base/cpu/cpu_context_impl.cc:155]: Does the empty buffer allow? If not maybe the FALT log level is better.

Comment (wyzero) [reply tao_compiler/mlir/xla/ral/context/base/cpu/cpu_context_impl.cc:155]: Yes, empty tensor is valid. Numpy (TF or Pytorch) supports empty tensor.

File: tao_compiler/mlir/disc/tests/regression/empty_tensor.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/tests/regression/empty_tensor.cc:57]: Yes, empty tensor is valid. Numpy (TF or Pytorch) supports empty tensor. For example,  we may use empty tensor as init value and expand the tensor (e.g. use concat op) by one in each iteration when doing decoding.

**Review by minminsun — APPROVED** [review]
LGTM
